security/pkg/k8s/controller/workloadsecret.go

```diff
@@ -41,6 +41,15 @@ const (
 	// The Istio secret annotation type
 	IstioSecretType = "istio.io/key-and-cert"
 
+	// NamespaceManagedLabel and NamespaceOverrideLabel contribute to determining whether or not
+	// a given Citadel instance should operate on a namespace. The behavior is as follows:
+	// 1) If NamespaceOverrideLabel exists and is valid, follow what this label tells us
+	// 2) If not, check NamespaceManagedLabel. If the value matches the Citadel instance's NS, then it should be active
```

Comment:
@costinm will this work now that we may do the one-namespace plan? We will have the same problem with the sidecar injector though

@Monkeyanator we are/might change the new install to have canaries be in the same namespace, so you would have citadel-foo and citadel-bar both in istio-system. So to identify them you would probably need a hostname instead of namespaces. Not sure exactly how this will work though, and whatever we do will need to be in sync with the injector

Reply:
Right, we were discussing using an identifier that would let us distinguish between Citadel instances in the same namespace, but from asking around, we found that the semantics of this hadn't been worked out yet, so we opted to do what the sidecar injector is doing right now

Reply:
I think I have a solution that will work with same-namespace canary. 

In this PR (or separate one) - can you add an extra env to control the prefix (currently hardcoded 'istio.') ?

Reply:
i.e. we can have a 'citadel canary' version that may generate different version of secrets ( with DNS names for canary injector, maybe different options) - and it will not interfere with the prod ( while still using the same root CA)